### PR TITLE
HLE Audio: Increase frame position by input buffer sample rate

### PR DESCRIPTION
**Pull request by jroweboy**
Currently the frame position moves ahead by the number of samples
output, but thats a fixed number based on the 3ds native sample rate.
Instead, based on a homebrew by cyuubi and looking at the lle audio,
this sample position should be moved forward by the number of samples
from the input buffer that was read, based on the buffer's sample rate.

This might fix the following issues related to audio playback/sync in citra 

- https://github.com/citra-emu/citra/issues/4649 
- https://github.com/citra-emu/citra/issues/5045 
- less likely, but it might fix games with issues in HLE but not LLE (won't affect games that don't work in LLE)

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5049)

### CODE REVIEW

**Comment by Cyuubi** [comment]
Source code: [ndsp-aud.zip](https://github.com/citra-emu/citra/files/4022822/ndsp-aud.zip)
Compiled binary: [ndsp-aud.zip](https://github.com/citra-emu/citra/files/4022823/ndsp-aud.zip)

Place a WAV file under 30 MB into your SDMC, enjoy? The first part is the audio time, second being OS time calculated from when the audio playback starts.

File: src/audio_core/hle/source.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2016 Citra Emulator Project
+﻿// Copyright 2016 Citra Emulator Project
```
Comment (wwylele) [inline src/audio_core/hle/source.cpp]: 🤔 what's this diff

Comment (Cyuubi) [reply src/audio_core/hle/source.cpp]: I was wondering the same thing.

Comment (jroweboy) [reply src/audio_core/hle/source.cpp]: I'm *pretty* sure thats my windows is adding a BOM to the file :P https://en.wikipedia.org/wiki/Byte_order_mark

Comment (wwylele) [reply src/audio_core/hle/source.cpp]: Please remove it then.

**Pull request merged**